src/intertypes/InterTypes.jl

```diff
@@ -113,6 +116,36 @@ function generate_module(mod::Module, target::Type{<:ExportTarget}, path="."; ta
   generate_module(mod.Meta, target, path; target_specific_args...)
 end
 
+module InterTypeSupport
+using OrderedCollections
+using StructEquality
+import StructTypes
+export Object, Optional
+
+@struct_hash_equal struct Object{T}
+  fields::OrderedDict{Symbol, T}
```

Comment:
Is this supposed to require that all the fields have same type?

Reply:
Yes. We don't support dynamic types. If you want fields of different types, use a struct where each field can be configured individually.

Reply:
ok, so you can have dynamic names where all the fields are the same type. Or static names with configurable types (also statically)